server/main_server/src/rodsServer.cpp

```diff
@@ -2457,14 +2457,20 @@ void readWorkerTask()
 
             while (bytes_to_send) {
                 const int bytes_sent = send(newSock, &(heartbeat[heartbeat_length - bytes_to_send]), bytes_to_send, 0);
-                const int errsav = errno;
-                if (bytes_sent > 0) {
-                    bytes_to_send -= bytes_sent;
+
+                // We do not check for 0 because it is not considered an error condition. It only means 0
+                // bytes were sent. As long as we have more bytes to send and no error has occurred, we
+                // keep attempting to send bytes.
+                if (bytes_sent == -1) {
+                    const int errsav = errno;
+                    if (errsav != EINTR) {
+                        log_server::error(
+                            "Socket error encountered during heartbeat; socket returned {}", strerror(errsav));
```

Comment:
main has this logged to the `agent` category: https://github.com/irods/irods/commit/0697ba2a02e5d89d5fa873e894f9f4dcadab7538 Is this because the heartbeat is not being sent to an agent in 4-3-stable?

Reply:
I think this happens before an agent is ever forked in iRODS 4.

Reply:
Got it. Thanks.